tasks/authorize.yml

```diff
@@ -4,9 +4,10 @@
     user: "postgres"
     key: "{{ backup_server_pub_key['content']|b64decode }}"
     state: present
-  delegate_to: "{{ hostvars[item].ansible_host | default(item) }}"
-  with_items: "{{ groups[barman_database.ansible_group] }}"
+  delegate_to: "{{ item }}"
+  with_items: "{{ groups[barman_database.ansible_group|default('')] | default(barman_database.authorize_ssh_hosts) }}"
```

Comment:
I don't understand why you have removed the reference to `ansible_group` in the configuration example, but keep it there. And I can't see why we should make a difference between "authorize_ssh_hosts" and "all_hosts".

If we have multiple standbys and want barman to be able to use any of them in case of failover, why not:
- have the `primary_host` configuration parameter (mandatory), nothing to do there
- add a `standby_hosts` configuration

And then in the role, we build a `all_hosts` variable, that is the concatenation of `primary_host` and `standby_hosts`, and on that list, we:
- add the public SSH key of the barman server
- add them to the `~/.pgpass` file

What do you think?

Reply:
that's is the main point of this PR: having the ability to differentiate the way we connect to `standby_hosts`.

In the case of the `delegate_to` usage to authorize the SSH key of the barman server, we don't especially need a particular info about the host (some ansible inventory can expose only an internal name such as `db1`) if the host is defined in the inventory with this name then the delegation will work (and use ansible's ssh connection info - again described in the inventory host file or variables)

In the case of building the `.pgpass` file we need a list of hostnames or ip addresses depending on how your infrastructure is built. 

Imagine such an inventory:
`hosts` file
```
[dbs]
db1 ansible_user=admin ansible_host=172.1.0.2
db2 ansible_user=admin ansible_host=172.1.0.3
```

you can `delegate_to: 'db1'` and Ansible will automatically ssh to the host via `ssh admin@172.1.0.2`
However the host names in the inventory (here `db1` and `db2`) will **NOT WORK** in the `.pgpass` file.
If we try to use the list of IPs directly it will **NOT WORK** neither (`delegate_to: 172.1.0.2` will not work because Ansible will not match the SSH connection info for that host)

Reply:
I like your variable name more though `standby_hosts`. I'll use that :+1: